File: Languages/dotnet/src/provider/MaxValueAnalysis.cs
using System;
using inputclass.src.model;
using System.Collections.Generic;

namespace inputclass.src.provider {
    
    /// <summary>
    /// Classe que implementa a interface "SimpleTableAnalysis",
    /// responsável por obter o maior valor de "credit" em uma lista de usuarios
    /// </summary>
    public class MaxValueAnalysis : SimpleTableAnalysis {
    
        /// <summary>
        /// Método de interface, responsável por obter o maior valor de credit na lista de "UserInfo"
        /// </summary>
        /// <param name="list">Lista de "UserInfo"</param>
        /// <returns>Maio valor de credit</returns>
        public double analysis(List<UserInfo> list) {

            double max = double.MinValue;
            foreach(UserInfo userInfo in list) {
                if(max < userInfo.credit) max = userInfo.credit;
            }

            return max;
        }
    }
}

File: Languages/typescript/src/provider/MaxValueAnalysis.ts
import { SimpleTableAnalysis } from '../model/SimpleTableAnalysis'
import { UserInfo } from '../model/UserInfo';

// Classe de analise que implementa a interface "SimpleTableAnalysis"
export class MaxValueAnalysis implements SimpleTableAnalysis {
   
    // Método de interface, responsável por realizar obter o valor maximo de credit em uma lista de usuarios
    // @param userInfoList Lista de usuarios
    // @return Valor maximo de credit
    public analysis(userInfoList: Array<UserInfo>): number{

        var max: number = Number.MIN_VALUE;

        userInfoList.forEach(userInfo => {
            if(max < userInfo.getCredit()) max = userInfo.getCredit()
        });

        return max;
    }
}